File: TwoSum.java
import java.util.*;
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        HashMap<Integer,Integer> map = new HashMap<>();
        
        for(int i=0;i<nums.length;i++)
        {
            int complement=target-nums[i];
            if(map.containsKey(complement))
            {
                   // ans[0]=map.get(nums[i]);
                    //ans[1]=i;
                return new int[] { map.get(complement),i };
                
            }
            map.put(nums[i],i);
            
        }
        
         throw new IllegalArgumentException("No two sum solution");
        
}
}

File: Python/TwoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        twosum_dict={}   #{element:index}
        
        for i,num in enumerate(nums):
            compliment = target - num
            if compliment in twosum_dict:
                return (i,twosum_dict[compliment])
            else:
                twosum_dict[num]=i
        
        
        